### PR TITLE
Lifecycle Update: Tracing

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

Removing mention of betas per https://datadoghq.atlassian.net/wiki/spaces/PM/pages/3310617880/Updated+Product+Lifecycle+Datadog.

Note: None of this stuff is tracked in the [beta tracker](https://docs.google.com/spreadsheets/d/1vZGzbad0wcnunYqsDV9CTPVt5nSkq8KtB0oKFOPE4Dc/edit?gid=773250897#gid=773250897), so I used my intuition for most of it. Happy to adjust anything!

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Comment by bm1549** [comment]
@brett0000FF  there are a few different areas that I would appreciate sign-off from different folks. I'll do my best to break down who should approve what below and put checkboxes for each, but the at-mentioned folks should review each of their respective sections or re-assign if needed

@bm1549 
- [ ] Sampling/ingestion control

@tushar-datadog 
- [ ] Config at runtime
- [ ] OpenTelemetry instrumentation (drop-in support)
- [ ] Span links APIs

@vidurkhannadatadog 
- [ ] Lib injection

@nevilgeorge
- [ ] Inferred services
- [ ] Service health

@delner 
- [ ] Orchestrion

@zacharycmontoya 
- [ ] .NET trimmed apps

@ancostas 
- [ ] Datadog connector for OpenTelemetry metrics

@alimimk 
- [ ] OpenAI configs for PHP

**Review by bm1549 — APPROVED** [review]
Approved for ingestion control

**Review by sabrenner — APPROVED** [review]
Reviewed the OpenAI changes in `content/en/tracing/trace_collection/library_config/php.md` on behalf of ML Observability, looks like it matches our Python spec that has been established in it's OpenAI integration, so should be good to remove the beta tags. Thanks!

**Review by labbati — COMMENTED** [review]
Thanks @brett0000FF, just added a few  quick comments, most of them are not blocking

File: content/en/tracing/guide/resource_based_sampling.md
Comment (brett0000FF) [reply content/en/tracing/guide/resource_based_sampling.md:14]: Thanks for the call out! I won't merge this until Websites sets up redirects for the new URL. 👍

File: content/en/tracing/trace_collection/compatibility/dotnet-core.md
Comment (brett0000FF) [reply content/en/tracing/trace_collection/compatibility/dotnet-core.md:185]: Great point! I agree it is worth updating the anchors since they display prominently in the URL. I forgot about that when I was going through this content initially. I'll fix that!

**Review by tushar-datadog — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]
Approved for "dotnet" changes and .NET trimmed apps

**Review by lee-brian — COMMENTED** [review]
Looks good to me!

**Review by hestonhoffman — APPROVED** [review]
Approving. Heads up: the redirects for these haven't been set up yet

File: content/en/tracing/trace_collection/library_config/dotnet-framework.md
Comment (lucaspimentel) [inline content/en/tracing/trace_collection/library_config/dotnet-framework.md:193]: This is a git merge conflict that needs to be resolved.

File: content/en/tracing/trace_collection/library_config/dotnet-core.md
Comment (lucaspimentel) [inline content/en/tracing/trace_collection/library_config/dotnet-core.md:176]: This is a git merge conflict that needs to be resolved.

**Pull request merged**
